src/EncoderDecoder/Huffman/HuffmanEncoderDecoder.cpp

```diff
@@ -0,0 +1,252 @@
+#include "EncoderDecoder/Huffman/HuffmanEncoderDecoder.h"
+
+HuffmanEncoderDecoder::NodeVector HuffmanEncoderDecoder::getFrequencies(
+	const std::string &plaintext) {
+
+	// Initialize frequencies
+	NodeVector frequencies;
+	frequencies.reserve(256);
+	for(int c = 0; c < 256; c++) {
+		frequencies.push_back(std::make_unique<Node>(Node(c)));
+	}
+
+	// Count frequency of each char in plaintext
+	std::for_each(plaintext.cbegin(), plaintext.cend(), [&](unsigned char c) {
+		auto &fp = frequencies[c];
+		(*fp)++;
```

Comment:
It doesn’t matter here, but I think it’s better to prefer the prefix ++ over the postfix one

Reply:
This requires adding another function to the `Node` interface which isn't important enough right now, so I'll pass.